### ISSUE TITLE
Cloud Source Unavailable

### ISSUE THREAD

**Issue by hdospitia** [GoogleCloudPlatform/pbmm-on-gcp-onboarding#469]
The repo uses Cloud Source by default, but this API was removed for new customers, effective June 17, 2024. For this reason, the bootstrap-0 step fails (TF apply output):

`Error: Request `Enable Project Service "sourcerepo.googleapis.com" for project "prj-b-cicd-rsoy"` returned error: failed to enable services: failed on request preconditions: googleapi: Error 403: Permission denied to enable service [sourcerepo.googleapis.com]`

![image](https://github.com/user-attachments/assets/c9a536f2-0165-4f70-8cd2-0e29ae755d50)

https://cloud.google.com/source-repositories/docs/reference/rest/

A change of repository resource should be designed.

**Comment by mromascanu123** [comment]
In fact looking at the code I see in 0-bootstrap a prep.sh script which selects the deployment metchd by copying the relevant files from the builders folder

From what I see the only deployment methods supported are "local" (i.e. running the deployment scripts from a Linux machine) and the pretty much equivalent devops pipeline (under azure-pipelines) which invokes the scripts under automation-scripts and reproduces the manual steps in a devops agent

**Comment by tackaberry** [comment]
This is correct. Cloud Soure Respositories will no longer be available and an alternate path should be chosen.   The GitHub version including GH automation is viable and the method is very similar to that of CSR.  I'm aware of efforts from the TEF team, as part of v5 ([see this issue](https://github.com/terraform-google-modules/terraform-example-foundation/issues/1249#issuecomment-2126684738)),  to build a methodology that uses Github repo with Cloudbuild automation.   We'll reintegrate and use that when its available .  

As @mromascanu123 said, the code under azure pipelines provide an alternative method to use Azure DevOps as code repo and automation.
